src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt

```diff
@@ -128,6 +127,51 @@ class SnykBulkFileListener : BulkFileListener {
         }
     }
 
+    private fun updateIacCache(
+        virtualFilesAffected: Set<VirtualFile>,
+        project: Project,
+        toolWindowPanel: SnykToolWindowPanel
+    ) {
+        val iacCache = toolWindowPanel.currentIacResult
+        if (iacCache?.allCliIssues == null) return
+        val iacFiles = iacCache.allCliIssues!!
+
+        ProgressManager.getInstance().run {
```

Comment:
ProgressManager#run(Task.Backgroundable) should be called here 
search in project `Task.Backgroundable` for example usage

Reply:
com.intellij.openapi.progress.ProgressKt#runBackgroundableTask will fit perfectly here